tests/scripts/branch-history

```diff
@@ -0,0 +1,209 @@
+#!/bin/bash
+
+# Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at the
+# Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights reserved.
+# See file COPYRIGHT for details.
+#
+# This file is part of the MFEM library. For more information and source code
+# availability see http://mfem.org.
+#
+# MFEM is free software; you can redistribute it and/or modify it under the
+# terms of the GNU Lesser General Public License (as published by the Free
+# Software Foundation) version 2.1 dated February 1999.
+
+# Print usage information
+case $1 in
+   -h|-help)
+      cat <<EOF
+
+   $0 [-h|-help] {mfem_dir} [-commit <hash>] [-range <hash>..<hash>] [-fast]
+
+   where: {mfem_dir}  is the MFEM source directory [default value: ../..]
+          -h|-help    prints this usage information and exits
+
+   This script checks if the current branch history, defined as the commits that
+   will be merged in master (those shown in a GitHub PR), contains unusually
+   large files, unusually large number of changes in a commit, unusually large
+   number of commits, etc. (Optionally one can specify a different range of
+   commits to be checked via the -commit and -range options.)
+
+   The specific checks are provided at 3 levels: branch, commit and file.
+
+   Branch-level checks:
+   - Check if the branch contains unusually large number of commits
+   - Check if the branch adds/modifies unusually large files
+
+   Commit-level checks:
+   - Check if the commit changes an unusually large number of files
+   - Check if the commit changes an unusually large number of lines
+
+   File-level checks:
+   - Check if the file is binary
+   - Check if the file is unusually large
+   - Check if an unusually large number of lines have been added to the file
+   - Check if an unusually large number of lines have been deleted from the file
+
+   The option -fast, allows to skip the checks that require checkout of the
+   individual commits, which could be convenient when there are local
+   changes. Currently, only the first two file-level checks will be skipped.
+
+   To be useful, the script should be run in a clean repository that doesn't
+   have any uncommitted changes and is not in a "detached HEAD" state.
+
+   Example usage: $0 ../..
+                  $0 ../../ -commit 8d50f48
+                  $0 ../../ -range eb2017b..5dd5511
+                  $0 ../../ -range master..d5f6f7e
+                  $0 ../../ -commit d5f6f7e -fast > /dev/null
+
```

Comment:
Tests for bad commits and ranges

Reply:
Great!
I haven’t tested is but it means we could use is in interactive rebase:
`> git rebase -i master --exec tests/scripts/branch_history ./ -commit HEAD -fast`
Or something similar.